packages/core/src/routes/session.ts

```diff
@@ -54,7 +53,13 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
     koaGuard({ body: object({ username: string(), password: string() }) }),
     async (ctx, next) => {
       const { username, password } = ctx.guard.body;
-      await signInWithUsernameAndPassword(ctx, provider, username, password);
+      assertThat(username && password, 'session.insufficient_info');
+
+      const { id } = await findUserByUsernameAndPassword(username, password);
+      await assignInteractionResults(ctx, provider, { login: { accountId: id } });
+      ctx.userLog.userId = id;
+      ctx.userLog.username = username;
```

Comment:
this line need to be moved before `findUserByUsernameAndPassword`, then failed attempt can be logged.